model/lifecycle.go

```diff
@@ -1426,14 +1426,14 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 	taskIds := []string{}
 	for _, b := range builds {
 		// Find the set of task names that already exist for the given build
-		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey))
+		tasksInBuild, err := task.Find(task.ByBuildId(b.Id))
```

Comment:
If we're worried this will be too large, we could just get fields DisplayName and Activated, and only get the full task in the case that the task needs to be activated.

Reply:
I advocate the other way. This is a loop over every build in the version, getting every task for that build. It looks like I made this call, and it wasn't because we hit some bug in production, but my intuition is the same about it now as it was then https://github.com/evergreen-ci/evergreen/pull/754.